src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Value.ReadTests.cs

```diff
@@ -318,24 +320,52 @@ public static void ValueFail()
             Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<Enum>(unexpectedString));
         }
 
-        [Fact]
-        public static void ReadVersion()
+        [Theory]
+        [InlineData("1.2")]
+        [InlineData("\\u0031\\u002e\\u0032", "1.2")]
+        [InlineData("1.2.3")]
+        [InlineData("\\u0031\\u002e\\u0032\\u002e\\u0033", "1.2.3")]
+        [InlineData("1.2.3.4")]
+        [InlineData("\\u0031\\u002e\\u0032\\u002e\\u0033\\u002e\\u0034", "1.2.3.4")]
+        [InlineData("2147483647.2147483647")]
+        [InlineData("\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037",
+            "2147483647.2147483647")]
+        [InlineData("2147483647.2147483647.2147483647")]
+        [InlineData("\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037",
+            "2147483647.2147483647.2147483647")]
+        [InlineData("2147483647.2147483647.2147483647.2147483647")]
+        [InlineData("\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037\\u002e\\u0032\\u0031\\u0034\\u0037\\u0034\\u0038\\u0033\\u0036\\u0034\\u0037",
+            "2147483647.2147483647.2147483647.2147483647")]
+        public static void Version_Read_Success(string json, string? actual = null)
         {
-            Version version;
-
-            version = JsonSerializer.Deserialize<Version>(@"""1.2""");
-            Assert.Equal(new Version(1, 2), version);
+            actual ??= json;
+            Version value = JsonSerializer.Deserialize<Version>($"\"{json}\"");
 
-            version = JsonSerializer.Deserialize<Version>(@"""1.2.3""");
-            Assert.Equal(new Version(1, 2, 3), version);
-
-            version = JsonSerializer.Deserialize<Version>(@"""1.2.3.4""");
-            Assert.Equal(new Version(1, 2, 3, 4), version);
+            Assert.Equal(Version.Parse(actual), value);
+        }
 
-            version = JsonSerializer.Deserialize<Version>("null");
-            Assert.Null(version);
+        [Theory]
+        [InlineData("")]
+        [InlineData("     ")]
+        [InlineData(" ")]
+        [InlineData("2147483648.2147483648.2147483648.2147483648")] //int.MaxValue + 1
+        [InlineData("2147483647.2147483647.2147483647.21474836477")] // Slightly bigger in size than max length of Version
+        [InlineData("-2147483648.-2147483648")]
+        [InlineData("-2147483648.-2147483648.-2147483648")]
+        [InlineData("-2147483648.-2147483648.-2147483648.-2147483648")]
+        [InlineData("1.-1")]
```

Comment:
nit: add a test case, just "1" as an input that is invalid because it is "too small"

Reply:
Added test case with "1" as input in f5b6e71